documentation/v&v/V&V.tex

```diff
@@ -0,0 +1,588 @@
+The project, Book Bazar is a website that addresses these issues. McMaster community members may buy/sell textbooks with ease using Book Bazar.
+\section{Roadmap}
+\label{sec: roadmap}
+Using the milestones, listed in Section 6 of the Development Process document, the team aims to have Book Bazar completed by the week of March 6, a week prior to the team's f\textbf{}inal presentation.
+
+\section{Testing Plan}
+Integration testing was used for all top-level modules which transitively tests all bottom level modules. This style of testing was chosen because it effectively tests how users will interact with the system instead of implementation details. Unit tests were not used as in practice they are often burdensome to write and maintain while providing minimal benefit in many cases due to much of the system being mocked out.
+
+\subsection{Testing Tools}
+The team will be use the following tools for implementing tests:\\
+\bi
+    \item Cypress -- used to run all integration tests (API and UI).
+    \item Jest -- used for writing unit tests.
+    \item Mock Service Workers -- used for mocking out external service APIs.
+\ei
+\subsection{API Tests}
+In the design specification the behaviour of HTTP handler modules was defined. This behaviour was used to define an expected output for a variety of inputs. This required creating tooling to build and teardown a testing database.
+
+A serverless architecture was used for API execution which means that horizontal scaling of the system is handled by \href{https://vercel.com}{Vercel}, the hosting provider. Therefore, \textbf{load testing} was not performed.
+
+
+\subsection{UI Tests}
+UI tests have been created for each way a user can interact with a page.
+
+
+\subsection{PASS/FAIL Criterion}
+The criterion used to determine PASS/FAIL of the the test cases id given below:
+\begin{center}
+    {$actual\_output == expected\_output$}
+\end{center}
+
+
+
+\section{Traceability Matrix}
+Below tables go over the coverage of our test plan. Modules were tested multiple times and ensured that the Function Requirements implemented are correct. 
+\subsection{UserInterface Module (M1)}
+\begin{table}[H]
+\flushleft
+\begin{tabular}{|c|p{8.5cm}|c|}
+\hline
+ \rowcolor{lightgray} 
+\textbf{Test ID} &\textbf{Description} &\textbf{Corresponding Requirement}\\
+\hline
+M1T1 & should allow a user to login & FR11, FR15 \\
+\hline
+M1T2 & should allow a user to logout & FR16 \\
+\hline
+M1T3 & should allow a user to delete their account & FR12 \\
+\hline
+M1T4 & should allow a user to change their name & FR17 \\
+\hline
+M1T5 & should allow a user to change their profile picture & FR17 \\
+\hline
+M1T6 & should allow a user to delete their posting from account page & FR2 \\
+\hline
+M1T7 & should allow a user to edit their posting from account page & FR19 \\
```

Comment:
We don't yet have a numbered FR 19; this is something we need to do for the requirements rev1. What did we agree we would do for this? We should change either this or my post to be consistent.

Reply:
Replaced "FR19" with, "(Pending changes to functional requirements)"